tm2/pkg/sdk/auth/ante.go

```diff
@@ -139,11 +139,19 @@ func NewAnteHandler(ak AccountKeeper, bank BankKeeperI, sigGasConsumer Signature
 		stdSigs := tx.GetSignatures()
 
 		for i := 0; i < len(stdSigs); i++ {
+			var isNewAccount bool
 			// skip the fee payer, account is cached and fees were deducted already
 			if i != 0 {
 				signerAccs[i], res = GetSignerAcc(newCtx, ak, signerAddrs[i])
+				// only create new account when tx is a sponsor transaction
 				if !res.IsOK() {
-					return newCtx, res, true
+					if tx.IsSponsorTx() {
+						isNewAccount = true
```

Comment:
sponsored tx is not only for new accounts

Reply:
i'm not sure i'm understanding your idea, could you suggest more details ?

Reply:
@moul Actually, the logic here is: when the signer account is not found in the storage (the account receiving the sponsor has not existed on-chain before), we check if the transaction being processed is a sponsor transaction. Only then do we allow the creation of this new account on-chain instead of returning an error. This way, we can handle cases where the account does not exist but can still perform transactions through a sponsor transaction